### PR TITLE
Remove the option to start TLS without any server certificate.

### PR DESCRIPTION
**Pull request by royjacobson**

### CODE REVIEW

File: src/facade/dragonfly_listener.cc [hunk 1]
```diff
@@ -56,23 +56,14 @@ using absl::GetFlag;
...
-    SSL_CTX_set_security_level(ctx, 0);
-
-    // you can still connect with redis-cli with :
-    // redis-cli --tls --insecure --tls-ciphers "ADH:@SECLEVEL=0"
-    LOG(WARNING) << "tls-key-file not set, no keys are loaded and anonymous ciphers are enabled. "
-                 << "Do not use in production!";
+    LOG(ERROR)
+        << "To use TLS, a server certificate must be provided with the --tls_cert_file flag!";
+    exit(-1);
   } else {  // tls_key_file is set.
```
Comment (kostasrim) [inline src/facade/dragonfly_listener.cc:67]: Sine the body of the if-statement unconditionally exits, you can remove the else statement below :)

Comment (kostasrim) [inline src/facade/dragonfly_listener.cc:67]:
```suggestion
  // tls_key_file is set.
```

**Comment by kostasrim** [comment]
@romange I recall that you mentioned there was a reason for this execution path, can you confirm if my memory is correct here? If not plz accept.

File: src/facade/dragonfly_listener.cc [hunk 2]
```diff
@@ -56,23 +56,14 @@ using absl::GetFlag;
 namespace {
 
 #ifdef DFLY_USE_SSL
-// To connect: openssl s_client  -cipher "ADH:@SECLEVEL=0" -state -crlf  -connect 127.0.0.1:6380
-static SSL_CTX* CreateSslCntx() {
+SSL_CTX* CreateServerSslCntx() {
```
Comment (romange) [inline src/facade/dragonfly_listener.cc:59]: can you please adjust the explanation on how to connect to dragonfly with openssl cli client?

**Comment by romange** [comment]
@royjacobson  is it still relevant?

**Comment by royjacobson** [comment]
> @royjacobson is it still relevant?

Yeah

**Review by kostasrim — APPROVED** [review]

**Pull request merged**